Adding navigator.requestMIDIAccess page (#13212)
mdn/content: files/en-us/web/api/navigator/requestmidiaccess/index.md

---
title: Navigator.requestMIDIAccess()
slug: Web/API/Navigator/requestMIDIAccess
tags:
  - API
  - Web MIDI
  - Reference
  - Method
  - Navigator
  - Secure context
  - requestMIDIAccess
browser-compat: api.Navigator.requestMIDIAccess
---

{{DefaultAPISidebar("Web MIDI API")}}

> The **`Navigator.requestMIDIAccess()`** method

Author: Which bit are you referring to @jpmedley - the sidebar?

> The **`requestMIDIAccess()`** method of the {{domxref('Navigator')}} interface.

Author: ```suggestion
The **`requestMIDIAccess()`** method of the {{domxref('Navigator')}} interface
```

> The **`requestMIDIAccess()`** method of the {{domxref('Navigator')}} interface
> returns a {{jsxref('Promise')}} representing a request for access to MIDI devices on the user's system. This method is part of the [Web MIDI API](/en-US/docs/Web/API/Web_MIDI_API) which defines a means for enumerating, manipulating and accessing MIDI devices.
> 
> This method prompts the user for access to MIDI devices available to their system. If the user grants permission then the {{jsxref('Promise')}} resolves and a [`MIDIAccess`](/en-US/docs/Web/API/MIDIAccess) object is returned.

Reviewer: ```suggestion
The **`requestMIDIAccess()`** method of the {{domxref('Navigator')}} interface
returns a {{jsxref('Promise')}} representing a request for access to Musical Instrument Digital Interface (MIDI) devices on a user's system. This method is part of the [Web MIDI API](/en-US/docs/Web/API/Web_MIDI_API).

This method prompts the user for access to MIDI devices available to their system. If the user grants permission, the {{jsxref('Promise')}} resolves and a [`MIDIAccess`](/en-US/docs/Web/API/MIDIAccess) object is returned. This method provides a means for accessing, enumerating, and manipulating MIDI devices.
```

- **acronym**: I've expanded MIDI since this is its first occurrence on this page.
- **which**: I've suggested a split of the second sentence to avoid ambiguity about what 'which' is referring to. I've also moved part of the split sentence to second paragraph.
- **word usage**: Please check if "provides" conveys the intended meaning better than "defines".
- **list order and comma**: minor suggestion for updating list to "accessing, enumerating, and manipulating" and added comma before the last 'and' in in the list

Author: * I think we keep 'Musical Instrument Digital Interface' for the api landing page (https://developer.mozilla.org/en-US/docs/Web/API/Web_MIDI_API). It makes the explanation here very convoluted.
* The API provides the means for enumerating etc the MIDI devices - but I don't want to repeat 'Web MIDI API' so suggestion for this is welcome
* It provides - something is returned by the method 👍 
* Yeh we do do a comma before 'and' I don't like it tho 😆

Reviewer: > * I think we keep 'Musical Instrument Digital Interface' for the api landing page (https://developer.mozilla.org/en-US/docs/Web/API/Web_MIDI_API). It makes the explanation here very convoluted.
>
👍  
>
>  * The API provides the means for enumerating etc the MIDI devices - but I don't want to repeat 'Web MIDI API' so suggestion for this is welcome
> 
In that case, your original sentence was correct! Sorry for this confusion. Only a comma would be needed before 'which':
"This method is part of the [Web MIDI API](/en-US/docs/Web/API/Web_MIDI_API), which defines a means for enumerating, manipulating, and accessing MIDI devices."

If we all are to stick to the same style guide (Oxford), I guess it would be nice to have the comma before the 'and' 🙂.
>
>
>     * It provides - something is returned by the method 👍
> 
>     * Yeh we do do a comma before 'and' I don't like it tho 😆

Hope you'll be adding the comma after "If the user grants permission"

## Syntax

```js
navigator.requestMIDIAccess();
navigator.requestMIDIAccess(MIDIOptions);
```

### Parameters

- `MIDIOptions` {{optional_inline}}
  - : An {{jsxref('Object')}} representing options to pass into the method. These options are:
    - `sysex`

>       - : A {{jsxref('Boolean')}} value which, if set to `true`, allows the ability to send and receive system exclusive (sysex) messages. Default is `false`.

Reviewer: ```suggestion
      - : A {{jsxref('Boolean')}} value that, if set to `true`, allows the ability to send and receive system exclusive (sysex) messages. The default value is `false`.
```

    - `software`

>       - : A {{jsxref('Boolean')}} value which, if set to `true`, allows the system to utilize any installed software synthesizers. Default is `false`.

Reviewer: ```suggestion
      - : A {{jsxref('Boolean')}} value that, if set to `true`, allows the system to utilize any installed software synthesizers. The default value is `false`.
```

### Return value

A {{jsxref('Promise')}} that resolves with a [`MIDIAccess`](/en-US/docs/Web/API/MIDIAccess) object.

### Exceptions

- `AbortError`
  - : If the document or page is closed due to user navigation.
- `InvalidStateError`

>   - : If the underling system raises any errors.

Reviewer: Do you mean 'underlying'?

- `NotSupportedError`
  - : If the feature or options are not supported by the system.
- `SecurityError`

>   - : If the user or system denies the application from creating a [MIDIAccess](/en-US/docs/Web/API/MIDIAccess) object with the requested options, or if the document is not allowed to use the feature (For example in the case of an iframe without the correct [feature policy](/en-US/docs/Web/HTTP/Feature_Policy)).

Author: ```suggestion
  - : If the user or system denies the application from creating a [MIDIAccess](/en-US/docs/Web/API/MIDIAccess) object with the requested options, or if the document is not allowed to use the feature (for example in the case of an iframe without the correct [feature policy](/en-US/docs/Web/HTTP/Feature_Policy)).
```

>   - : If the user or system denies the application from creating a [MIDIAccess](/en-US/docs/Web/API/MIDIAccess) object with the requested options, or if the document is not allowed to use the feature (for example in the case of an iframe without the correct [feature policy](/en-US/docs/Web/HTTP/Feature_Policy)).

Reviewer: ```suggestion
  - : If the user or system denies the application from creating a [MIDIAccess](/en-US/docs/Web/API/MIDIAccess) object with the requested options, or if the document is not allowed to use the feature (for example, an iframe without the correct [feature policy](/en-US/docs/Web/HTTP/Feature_Policy)).
```

## Examples

> In the following example the {{domxref("Navigator.requestMIDIAccess()")}} method returns the {{domxref("MIDIAccess")}} object, which gives access to information about the input and output MIDI ports.

Reviewer: ```suggestion
In the following example, the {{domxref("Navigator.requestMIDIAccess()")}} method returns the {{domxref("MIDIAccess")}} object, which gives access to information about the input and output MIDI ports.
```
